App10.6/App10.6/10(6).cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+internal class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine("Введите начало промежутка a: ");
+        int a = int.Parse(Console.ReadLine());
+
+        Console.WriteLine("Введите конец промежутка b, большее a: ");
+        int b = int.Parse(Console.ReadLine());
+
+        if (a >= b)
+        {
+            Console.WriteLine("Ошибка: a должно быть меньше b.");
+            return;
+        }
+
+        Console.WriteLine("Введите сумму делителей n: ");
+        int n = int.Parse(Console.ReadLine());
+
+        Console.WriteLine($"Числа от {a} до {b}, у которых сумма делителей равна {n}:");
+
+        bool found = false;
+
+        for (int num = a; num <= b; num++)
+        {
+            int sumOfDivisors = 0;
+
+            for (int i = 1; i <= num / 2; i++)
+            {
+                if (num % i == 0)
+                {
+                    sumOfDivisors += i;
+                }
+            }
```

Comment:
Перебор делителей неверный.

> Введите начало промежутка a:
1
Введите конец промежутка b, большее a:
20
Введите сумму делителей n:
12
Числа от 1 до 20, у которых сумма делителей равна 12:
Таких чисел нет

Однако такое число есть. Это 6. Его делители 1, 2. 3. 6. Сумма 1 + 2 +3 + 6 = 12.
Вы почему-то перебираете делители только до `num / 2`.  А надо до `num`. И снова некоторые скобки можно не писать, что облегчает чтение кода
```suggestion
            for (int i = 1; i <= num; i++)
                if (num % i == 0)
                    sumOfDivisors += i;
```
А если хотите сократить перебор, то так
```suggestion
            var sqrt = (int)Math.Sqrt(num);
            
            for (int i = 1; i < sqrt; i++)
                if (num % i == 0)
                    sumOfDivisors += i + num / i;
            
            if( sqrt * sqrt == num) // num является квадратом целого числа
                sumOfDivisors += sqrt;
            
```